packages/vue-client/src/components/PlayingTable/SFractionalTable.vue

```diff
@@ -1,15 +1,16 @@
 <script setup lang="ts">
 import {
+  DefaultBoardConfig,
+  DefaultBoardState,
   SFractionalConfig,
   sFractionalMoveColors,
-  SFractionalState,
 } from "@ogfcommunity/variants-shared";
 import MoveSequenceDisplay from "../MoveSequenceDisplay.vue";
-import SFractionalBoardSelector from "../boards/SFractional/SFractionalBoardSelector.vue";
+import DefaultBoard from "../boards/DefaultBoard.vue";
 
 const props = defineProps<{
-  gamestate: SFractionalState;
-  config: SFractionalConfig;
+  gamestate: DefaultBoardState;
+  config: SFractionalConfig & DefaultBoardConfig; // return type of SFractional.uiTransform
```

Comment:
Okay as-is, but TypeScript's `ReturnType` can make the comment something the compiler will understand.

https://www.typescriptlang.org/docs/handbook/utility-types.html#returntypetype

Reply:
When I try `ReturnType<typeof sFractionalVariant.uiTransform>` I get the error
```
Type '((config: SFractionalConfig, state: SFractionalState) => { config: DefaultBoardConfig; gamestate: DefaultBoardState; }) | undefined' does not satisfy the constraint '(...args: any) => any'.
  Type 'undefined' is not assignable to type '(...args: any) => any'.ts-plugin(2344)
```
Is there a way to assert that it's not undefined?

Reply:
Right, Variant.uiTransform is gullable. But you can do `ReturnType<typeof SFractional.uiTransform>` (the static method) which should be guaranteed to exist

Reply:
Oh yeah, I missed that, thank you! 😄 
With a structure like this, I feel like we are quite type safe.